### ISSUE TITLE
Add pandas .dropna equivalent

### ISSUE THREAD

**Issue by Gabriel-p** [astropy/astropy#7446]
There is apparently no easy way to drop rows with masked values (as explained in https://stackoverflow.com/q/50256012/1391441).

A `dropna` equivalent should be added.

**Comment by pllim** [comment]
For now... maybe this?
```python
>>> df = t.to_pandas()
>>> df.dropna()
>>> t2 = Table.from_pandas(df)
```

**Comment by Gabriel-p** [comment]
Yes, but that forces me to have `pandas` installed. I want my package to depend on the smallest number of packages possible.

**Comment by moon05** [comment]
@Gabriel-p Hi! Are you taking care of this? I was just curious if pretty much having a function with what literally one of the SO answers are stating would work? Since functools and itertools are already imported in `ascii` that seems like the easiest way to go?

**Comment by bsipocz** [comment]
@moon05 - I'm sure @pllim is not working actively on this, you're most welcome to open a PR with a prototype solution and get the discussion going in there.

**Comment by moon05** [comment]
@bsipocz thanks! I will open a PR tonight hopefully.
